packages/shared/lib/wallet.ts

```diff
@@ -586,6 +586,28 @@ export const asyncCreateAccount = (): Promise<void> =>
         )
     })
 
+export const asyncRemoveWalletAccount = (accountId: string): Promise<void> =>
+    new Promise<void>((resolve, reject) => {
+        api.removeAccount(accountId, {
+            onSuccess() {
+                /**
+                 * CAUTION: If an account is successfully removed in wallet.rs then it should also be
+                 * removed in the Firefly store. This is "inefficient" (esp. for batch deletes) but it
+                 * at least ensures data integrity / consistency between Firefly and the backend.
+                 */
+                get(wallet).accounts.update((_accounts) => _accounts.filter((wa) => wa.id !== accountId))
+
+                resolve()
+            },
+            onError(err) {
+                reject(err)
+            },
+        })
+    })
+
+export const asyncRemoveWalletAccounts = (accountIds: string[]): Promise<void[]> =>
+    Promise.all(accountIds.map((id) => asyncRemoveWalletAccount(id)))
+
```

Comment:
If we are deleting a whole profile, rather than deleting all accounts separately and doing the filter on the array many times; should we have an additional function to deleteAllWalletAccounts and then we just set the array to [] or undefined and we don't need to do all this computation.

Reply:
If this is added then it must be done library side on wallet.rs as an extra endpoint to `remove_accounts`, which a dev would only want to call if they were deleting profile / account manager. But if that's the case then they also need to call `delete` from the library to actually delete the storage... I think we should look at consolidating all this logic for deleting a profile into a single API call, that'd be nice.

Reply:
Okay lets add this to click up as a refactor for the future.